tasks/config-gitlab-runner.yml

```diff
@@ -8,10 +8,35 @@
     backrefs: True
   notify: restart gitlab-runner
   loop:
-    - {param: "concurrent", value: "{{ global_section.concurrent }}"}
+    - {param: 'concurrent', value: '{{ global_section.concurrent }}'}
   vars:
     global_section: "{{ gitlab_runner_config.global }}"
   become: True
 
 - name: Force all notified handlers to run
   meta: flush_handlers
+
+- name: List all configured runners
+  command: gitlab-runner list
+  register: configured_runners
+  failed_when: configured_runners.rc != 0
+  changed_when: False
+
+- name: register gitlab-runner
+  command: >-
+          gitlab-runner register
+          {% for env_var in gitlab_runner_env_vars %}
+          --env '{{ env_var }}'
+          {% endfor %}
+          {% if gitlab_runner_extra_options is defined %}
+          '{{ gitlab_runner_extra_options }}'
+          {% endif %}
```

Comment:
Jinja in tasks? Why?

Reply:
Because we need dynamically created command

Reply:
Good